src/test/java/cz/czechitas/TemplateTest.java

```diff
@@ -1,25 +1,69 @@
 package cz.czechitas;
 
-import org.junit.jupiter.api.AfterEach;
-import org.junit.jupiter.api.BeforeEach;
-import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.*;
+import org.omg.CORBA.DynAnyPackage.*;
+import org.openqa.selenium.*;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.support.ui.*;
+
+import static java.lang.Thread.*;
+import static org.junit.jupiter.api.Assertions.*;
 
 public class TemplateTest {
 
+    private static WebDriver driver;
+    private static ShopPage AdminUrlLogin;
+
+    private final int maxWaitInSeconds= 10;
 
     @BeforeEach
     public void before() {
-
+        driver = DriverHelper.openFirefox();
+        AdminUrlLogin = new ShopPage(driver);
+        AdminUrlLogin.openPage();
     }
 
     @AfterEach
     public void after() {
+        driver.quit();
+    }
+
+    @Test
+    @DisplayName ("Admin login failed, wrong password")
+    public void testAdminlogin1() {
+       AdminUrlLogin.ValidUsernameAdminLogin();
```

Comment:
Ty jsi trochu pomichala jak maji fungovat testy a page objecty. V page objectu ma byt od na interakci se strankou. V testu ma byt konkretni interakce se strankou.

Page Object:
```
public void login(String username, String password) {
        driver.findElement(usernameSelector).sendKeys(username);
        driver.findElement(passwordSelector).sendKeys(password);
        driver.findElement(loginButtonSelector).click();
    }
```

Test:
```
@Test
    @DisplayName("it sucessfully login")
    public void testValidLogin() {
        AdminPage adminPage = adminLoginPage.login(validUsername, validPassword);
        assertEquals(validUsername, adminPage.isLoggedInAs(), "User should be redirected to admin page");
    }
```

Dulezite je to, ze se vstupy do stranky vzdy posilaji z testu, a zaroven v testu nemaji byt uz zadne findElement a selectory. Popisu ti to jeste na konkretnich radcich.

Reply:
Moniko, ahoj, předělala jsem DÚ a pořádně oddělila PO a hlavní třídu pro test.